ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Run.hs

```diff
@@ -256,23 +270,28 @@ runPointSchedule schedulerConfig genesisTest tracer0 =
         -- the registry is closed and all threads related to the peer are
         -- killed.
         withRegistry $ \peerRegistry -> do
-          (csClient, csServer) <- startChainSyncConnectionThread peerRegistry tracer config chainDbView fetchClientRegistry prShared prChainSync chainSyncTimeouts_ chainSyncLoPBucketConfig stateViewTracers (psrCandidates resources) (psrHandles resources)
+          (csClient, csServer) <- startChainSyncConnectionThread peerRegistry tracer config chainDbView fetchClientRegistry prShared prChainSync chainSyncTimeouts_ chainSyncLoPBucketConfig stateViewTracers (psrCandidates resources) (psrHandles resources) (psrIdling resources)
           BlockFetch.startKeepAliveThread peerRegistry fetchClientRegistry (srPeerId prShared)
           (bfClient, bfServer) <- startBlockFetchConnectionThread peerRegistry tracer stateViewTracers fetchClientRegistry (pure Continue) prShared prBlockFetch blockFetchTimeouts_
           waitAnyThread [csClient, csServer, bfClient, bfServer]
     -- The block fetch logic needs to be started after the block fetch clients
     -- otherwise, an internal assertion fails because getCandidates yields more
     -- peer fragments than registered clients.
     let getCurrentChain = ChainDB.getCurrentChain chainDb
-        getPoints = traverse readTVar (srCurrentState . prShared <$> (psrPeers resources))
+        getPoints = traverse readTVar (srCurrentState . prShared <$> psrPeers resources)
         mkStateTracer
           | scTraceState schedulerConfig
           = peerSimStateDiagramSTMTracerDebug gtBlockTree getCurrentChain getCandidates getPoints
           | otherwise
           = pure nullTracer
+
+        gdd = updateLoEFragGenesis config (mkGDDTracerTestBlock tracer) getCandidates getHandles
+
     stateTracer <- mkStateTracer
     BlockFetch.startBlockFetchLogic registry chainDb fetchClientRegistry getCandidates
-    void $ forkLinkedThread registry "ChainSel trigger" (reprocessLoEBlocksOnCandidateChange chainDb getCandidates)
+    for_ loEVar $ \ var ->
+        void $ forkLinkedThread registry "LoE updater background" $
+          runGddAsync gdd var chainDb ((,) <$> getLatestSlots <*> readTVar (psrIdling resources))
```

Comment:
The GDD governor is not using `psrIdling` to compute densities. Is this going to be the subject of another PR?